app/routes/repos.py

```
@@ -51,3 +84,94 @@ async def get_repos(
+
+@router.get(
+    "/git_repos/{user_id}/{token_id}",
+    response_model=List[Dict[str, Any]],
+    status_code=status.HTTP_200_OK,
+    summary="Get all repos from provider",
+    description="Retrieve a paginated list of repositories based on provider sent for a user",
+)
+async def get_repos_from_git(
+    token_id: str = Path(...),
+    user_id: str = Path(...),
+    pagination: PaginationParams = Depends(),
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)